test-data/unit/check-python38.test

```diff
@@ -403,7 +403,14 @@ if (thing := get_thing()).is_good:
     reveal_type(thing)  # N: Revealed type is "__main__.Good"
 else:
     reveal_type(thing)  # N: Revealed type is "__main__.Bad"
-[builtins fixtures/property.pyi]
+
+def get_things() -> Union[Tuple[Good], Tuple[Bad]]: ...
+
+if (things := get_things())[0].is_good:
```

Comment:
You have modified two branches for `IndexExpr` and `MemberExpr`, but this new test covers `NameExpr`. Do we have other two covered?

Reply:
Good point, I'll fix this tomorrow. Although actually maybe we should do the walrus collapse inside the binder put

Reply:
The tomorrow has just started in my timezone 😆

Reply:
> this new test covers NameExpr

The new test actually covers IndexExpr in the branch above.

I tried adding tests with IndexExpr and MemberExpr inside the walrus, but it turns out that's not valid Python